### PR TITLE
Update cox_preflight_VE.R

### PR DESCRIPTION
**Pull request by elsie-h**

### CODE REVIEW

File: analysis/VE/cox_preflight_VE.R [hunk 1]
```diff
@@ -23,15 +23,15 @@ args <- commandArgs(trailingOnly=TRUE)
 
 ## Set input and output pathways for matched/unmatched data - default is unmatched
 # arg1: db = matched / unmatched
-# arg2: timescale = persontime / calendartime / calendartimefull
+# arg2: timescale = persontime / calendartime
```
Comment (elsie-h) [inline analysis/VE/cox_preflight_VE.R:26]: Drop the `calendartimefull` option, and always run the full and stratified models when `arg2=calendartime`.

File: analysis/VE/cox_preflight_VE.R [hunk 2]
```diff
@@ -193,8 +185,7 @@ postvax_time <- data_cohort %>%
 ####################################################### 
 ### Specify variables for models
 #######################################################
-surv_strata = c("tstart", "tstop", "ind_outcome") 
-surv_full = c("follow_up_time", "ind_outcome")
+surv = c("tstart", "tstop", "ind_outcome")
```
Comment (elsie-h) [inline analysis/VE/cox_preflight_VE.R:188]: use the format `Surv(tstart, tstop, ind_outcome)` for all models, with `tstart=0` for all persontime models

File: analysis/VE/cox_preflight_VE.R [hunk 3]
```diff
@@ -204,129 +195,134 @@ vars2_cat = c("sex", "imd", "ethnicity", "rural_urban_group", "ckd_3cat", "multi
...
 
-
-
 ####################################################### 
 ### Prepare time to event data for the given outcome
 ####################################################### 
-if (timescale == "persontime") {
+
+# TODO
+# ensure 85+ cut from boost model
```
Comment (elsie-h) [inline analysis/VE/cox_preflight_VE.R:208]: Need to make sure that the updated code is doing the right thing for boost models

File: analysis/VE/cox_preflight_VE.R [hunk 4]
```diff
@@ -204,129 +195,134 @@ vars2_cat = c("sex", "imd", "ethnicity", "rural_urban_group", "ckd_3cat", "multi
...
 }
 
 ## Matched models to be run without additional confounder adjustment
 
-
-
 ####################################################### 
 ### Prepare time to event data for the given outcome
 ####################################################### 
-if (timescale == "persontime") {
```
Comment (elsie-h) [inline analysis/VE/cox_preflight_VE.R:217]: `data_tte` same for both person and calendar time models, then rescale later for calendar time

File: analysis/VE/cox_preflight_VE.R [hunk 5]
```diff
@@ -204,129 +195,134 @@ vars2_cat = c("sex", "imd", "ethnicity", "rural_urban_group", "ckd_3cat", "multi
...
+    # )
+  )
+
+# define full data
+data_cox_full <- data_tte %>%
+  mutate(
+    tstart = 0,
+    tstop = tte_outcome
+  )
+
```
Comment (elsie-h) [inline analysis/VE/cox_preflight_VE.R:250]: rescaling for calendar time below

File: analysis/VE/cox_preflight_VE.R [hunk 6]
```diff
@@ -204,129 +195,134 @@ vars2_cat = c("sex", "imd", "ethnicity", "rural_urban_group", "ckd_3cat", "multi
...
+      id = patient_id,
+      timesincevax_pw = tdc(fup_day, timesincevax_pw)
+    ) %>%
+    # set factor levels for postvaccination periods
+    mutate(across(timesincevax_pw, factor, levels = postvax_periods)) %>%
+    # ind_outcome as logical to match data_tte
+    mutate(across(ind_outcome, as.logical)) %>%
+    as_tibble()
   
-  data_cox_full <- data_cox_strata
```
Comment (elsie-h) [inline analysis/VE/cox_preflight_VE.R:313]: rescale if calendar time

File: analysis/VE/cox_preflight_VE.R [hunk 7]
```diff
@@ -335,20 +331,18 @@ if (!full | (timescale == "calendartime" | timescale == "calendartimefull")) {
...
-} 
+}
 
-## Specify stratified dataset for all primary models fit to whole population, and booster models with calendertimefull implementation
-if ((vaccine=="primary" & subgroup=="all") | (vaccine=="boost" & timescale=="calendartimefull")) {
+## Specify stratified dataset
+if (strata) {
   ## Only keep postvaxcuts during which there are >2 events for each level of expo
   data_cox_strata_keep <- data_cox_strata %>%
-    filter(!is.na(timesincevax_pw) & timesincevax_pw!="183+" & timesincevax_pw!="85+") %>% ## Added to cut periods of 1-14d and 183d+ (primary) or 85+ (boost)
```
Comment (elsie-h) [inline analysis/VE/cox_preflight_VE.R:351]: Double check this is in the data_select censoring.

**Comment by elsie-h** [comment]
The [add region_jcvi strata to persontime formulas](https://github.com/opensafely/ckd-coverage-ve/pull/5/commits/1fa23610a2241b1b7996a2fffe175df0545f55cf) commit adds the JCVI and region strata to the Cox model formulas for the person time models, to align with the calendar time models. I previously defined :

`strata_var = as.character(glue("{jcvi_group}_{region}"))`

but I've changed this to 

`jcvi_region = as.character(glue("{jcvi_group}_{region}"))`

to avoid any confusion between the `strata` argument which controls whether the follow-up time is stratified.

File: analysis/VE/cox_preflight_VE.R (hunk 7 above)
Comment (eparker12) [reply analysis/VE/cox_preflight_VE.R:351]: Done. The booster data selection script contains the following, so censor date will be a maximum of 84 days after the booster dose.
`censor_date = pmin(vax3_date - 1 + lastfupday, end_date, dereg_date, death_date, vax4_date, na.rm=TRUE),`

**Review by eparker12 — APPROVED** [review]
Thanks - this is a much more coherent script thanks to these changes

**Review by eparker12 — APPROVED** [review]

**Pull request merged**